tutorials/plot_spatial.py

```diff
@@ -11,40 +11,237 @@
+# Make figure bigger
+f.set_size_inches((20/2.54, 10/2.54))
 
 ###################################################################################################
-# Subtitle
+# 3. Compute spatial bin assignment using spatial bin edges
 # ~~~~~~~~
 #
-# Words, words, words.
+# Compute spatial bin assignment for position using previously computed spatial bin edges.
 #
 
 ###################################################################################################
 
-# Do some code stuff
-print('Hello World!')
+# Now let us check where the 1st 7 point of position using the same x_edges and y_edges
+n_points = 7
+x_bins, y_bins = compute_spatial_bin_assignment(position[:, :n_points], x_edges, y_edges, include_edge=True)
+# We can check they match the positions in plot (ii)
+for i in range (0, n_points):
+    print(f'The point (x, y) = ({position[0, i]}, {position[1, i]}) is in the x_bin {x_bins[i]}, and on the y_bin {y_bins[i]}.')
+
+###################################################################################################
+# 4. Compute time in each timestamp sample
+# ~~~~~~~~
+#
+# Compute time width of the timestamp sampling bins.
+#
 
 ###################################################################################################
 
-# Do more code stuff
-print('More stuff!')
+# Let us now compute the time in each timestamp bin
+bin_time = compute_bin_time(timestamps)
+print(f'The time widths of the the sampling bins are: {bin_widths}')
 
 ###################################################################################################
-# Conclusion
-# ~~~~~~~~~~
+# 5. Compute and plot occupancy
+# ~~~~~~~~
 #
-# Words, words, words.
+# Compute occupancy using position, timestamps, bins and speed. Plot a heatmap of occupancy.
 #
+
+###################################################################################################
+
+# Compute occupancy using the previously defined position, timestamps, bins and speed
+occupancy = compute_occupancy(position, timestamps, bins, speed=np.insert(speeds, 0, 0))
+
+# Plot occupancy using a heatmap
+sns.heatmap(occupancy)
+plt.title('Occupancy heatmap')
+plt.xlabel('x spatial bin')
+plt.xlabel('y spatial bin')
+
+###################################################################################################
+# 6. Compute 2D and 1D spatial information
+# ~~~~~~~~
+#
+# Compute spatial information for simulated spike positions, spatial bins, and occupancy.
+#
+
+###################################################################################################
+
+# Simulate a spike train with chance level
+spike_train = sim_spiketrain_binom(0.5, n_samples=len(x_pos))
+# Get x and y positions corresponding
+spike_x = x_pos[np.where(spike_train == 1)]
+spike_y = y_pos[np.where(spike_train == 1)]
+
+###################################################################################################
+
+# Calculate the 1D spatial information (x-dimension only)
+data = spike_x
```

Comment:
I don't think you need to rename this variable here - you can just pass `spike_x` into the function directly